File: java/IEC61970/Base/Generation/Production/CombinedCyclePlant.java
package IEC61970.Base.Generation.Production;

import IEC61970.Base.Domain.ActivePower;
import IEC61970.Base.Core.PowerSystemResource;

/**
 * A set of combustion turbines and steam turbines where the exhaust heat from the
 * combustion turbines is recovered to make steam for the steam turbines,
 * resulting in greater overall plant efficiency.
 * @created 02-Jan-2024 10:52:09 PM
 */
public class CombinedCyclePlant extends PowerSystemResource {

	/**
	 * The combined cycle plant's active power output rating.
	 */
	public ActivePower combCyclePlantRating;
	/**
	 * A thermal generating unit may be a member of a combined cycle plant.
	 */
	public ThermalGeneratingUnit ThermalGeneratingUnits;

	public CombinedCyclePlant(){

	}
}//end CombinedCyclePlant

File: py/IEC61970/Base/Generation/Production/CombinedCyclePlant.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Sun Jan  7 12:46:14 2024
# The set of combustion turbines and steam turbines where the exhaust heat from the
# combustion turbines is recovered to make steam for the steam turbines,
# resulting in greater overall plant efficiency.
from IEC61970.Base.Core.PowerSystemResource import PowerSystemResource
from IEC61970.Base.Domain.ActivePower import ActivePower
from IEC61970.Base.Generation.Production.ThermalGeneratingUnit import ThermalGeneratingUnit


class CombinedCyclePlant(PowerSystemResource):
    """
    A set of combustion turbines and steam turbines where the exhaust heat from the
    combustion turbines is recovered to make steam for the steam turbines,
    resulting in greater overall plant efficiency.
    @created 02-Jan-2024 10:52:09 PM
    """

    def __init__(self):
        super().__init__()
        # The combined cycle plant's active power output rating.
        self.comb_cycle_plant_rating = ActivePower()  # Assuming ActivePower class has typical values as arguments
        # A thermal generating unit may be a member of a combined cycle plant.
        self.thermal_generating_units = ThermalGeneratingUnit()  # Assuming ThermalGeneratingUnit class has typical values as arguments
